packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -386,3 +401,536 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
+      return;
+    }
+    _fadeoutTimer?.cancel();
+    _fadeoutAnimationController.forward();
+    switch (direction) {
+      case Axis.vertical:
+        _dragScrollbarAxisPosition = details.localPosition.dy;
+        break;
+      case Axis.horizontal:
+        _dragScrollbarAxisPosition = details.localPosition.dx;
+        break;
+    }
+  }
+
+  /// Handler called when a currently active long press gesture moves.
+  ///
+  /// Updates the position of the child scrollable.
+  @protected
+  void handleLongPressMoveUpdate(LongPressMoveUpdateDetails details) {
+    final Axis? direction = getDirection();
+    if (direction == null) {
+      return;
+    }
+    switch(direction) {
+      case Axis.vertical:
+        _updateScrollPosition(details.localPosition.dy - _dragScrollbarAxisPosition!);
+        _dragScrollbarAxisPosition = details.localPosition.dy;
+        break;
+      case Axis.horizontal:
+        _updateScrollPosition(details.localPosition.dx - _dragScrollbarAxisPosition!);
+        _dragScrollbarAxisPosition = details.localPosition.dx;
+        break;
+    }
+  }
+
+  /// Handler called when a long press has ended.
+  ///
+  /// Finishes any active drag action on the child scroll view.
+  @protected
+  void handleLongPressEnd(LongPressEndDetails details) {
+    final Axis? direction = getDirection();
+    if (direction == null)
+      return;
+    final Offset pixelsPerSecond = details.velocity.pixelsPerSecond;
+    _maybeStartFadeoutTimer();
+    _dragScrollbarAxisPosition = null;
+    final double scrollVelocity = painter!.getTrackToScroll(
+      direction == Axis.vertical
+        ? pixelsPerSecond.dy
+        : pixelsPerSecond.dx
+    );
+    _drag?.end(DragEndDetails(
+      primaryVelocity: -scrollVelocity,
+      velocity: Velocity(
+        pixelsPerSecond: direction == Axis.vertical
+          ? Offset(0.0, -scrollVelocity)
+          : Offset(-scrollVelocity, 0.0),
+      ),
+    ));
+    _drag = null;
+    _currentController = null;
+  }
+
+  /// Updates the scrollbar thumb and it's animations based on the received
+  /// [ScrollNotification].
+  @protected
+  bool handleScrollNotification(ScrollNotification notification) {
+    final ScrollMetrics metrics = notification.metrics;
+    if (metrics.maxScrollExtent <= metrics.minScrollExtent)
+      return false;
+
+    if (notification is ScrollUpdateNotification ||
+      notification is OverscrollNotification) {
+      // Any movements always makes the scrollbar start showing up.
+      if (_fadeoutAnimationController.status != AnimationStatus.forward)
+        _fadeoutAnimationController.forward();
+
+      fadeoutTimer?.cancel();
+      painter!.update(notification.metrics, notification.metrics.axisDirection);
+    } else if (notification is ScrollEndNotification) {
+      if (_dragScrollbarAxisPosition == null)
+        _maybeStartFadeoutTimer();
+    }
+    return false;
+  }
+
+  /// Get the GestureRecognizerFactories used to detect gestures on the scrollbar
```

Comment:
This needs to say more about what gestures we're handling and what methods (that subclasses can override) are called when those gestures occur.

Reply:
If we decide to remove all of the protected methods, then subclasses won't be able to override these. Actually, in the changes we discussed today, I think all subclasses will have to implement their own gestures.